third_party/nvfuser/csrc/kernel_ir.cpp

```diff
@@ -157,6 +157,12 @@ Allocate::Allocate(
     TORCH_INTERNAL_ASSERT(alias->memoryType() == memory_type, "Invalid alias");
   }
 
+  // FIXME: there is a bug in lower_alias_memory.cpp that causes
+  // `NVFuserTest.FusionPredicateElimination6_CUDA` to fail if I simplify `5*2`
+  // into `10`
+
+  // size = simplifyExpr(size);
```

Comment:
I am seeing this weird generated code:
```CUDA
    // Alias Allocation - register
    auto& T3 = T2;
    #pragma unroll
    for(nvfuser_index_t i38 = 0; i38 < 5; ++i38) {
      int i193;
      i193 = i192 + i38;
      #pragma unroll
      for(nvfuser_index_t i39 = 0; i39 < 2; ++i39) {
        T3[(i38 + (5 * i39))]
          = T2[(i193 + (3 * i39))]
          + (float) 1.00000000000000000e+00;
      }
    }
```

Reply:
What is this? What's the fusion math?

Reply:
CUDA kernel:
```CUDA
__global__ void kernel1(Tensor<float, 2> T0, Tensor<float, 2> T4) {
  NVFUSER_DEFINE_MAGIC_ZERO
  int i85;
  i85 = T0.stride[1] * 5;
  float T1[10];
  #pragma unroll
  for(nvfuser_index_t i32 = 0; i32 < 2; ++i32) {
    int i84;
    i84 = T0.stride[0] * i32;
    int i107;
    i107 = 3 * i32;
    #pragma unroll
    for(nvfuser_index_t i33 = 0; i33 < 1; ++i33) {
      int i87;
      i87 = i84 + (i85 * i33);
      int i108;
      i108 = 5 * i33;
      int i109;
      i109 = i107 + i108;
      int i270;
      i270 = 3 - i108;
      #pragma unroll
      for(nvfuser_index_t i34 = 0; i34 < 5; ++i34) {
        int i56;
        i56 = i34 + nvfuser_zero;
        if ((i56 < i270)) {
          T1[(i109 + i34)]
            = T0[(i87 + (T0.stride[1] * i56))]
            + (float) 1.00000000000000000e+00;
        }
      }
    }
  }
  NVFUSER_UPDATE_MAGIC_ZERO
  float T2[10];
  #pragma unroll
  for(nvfuser_index_t i35 = 0; i35 < 2; ++i35) {
    int i128;
    i128 = 3 * i35;
    #pragma unroll
    for(nvfuser_index_t i36 = 0; i36 < 1; ++i36) {
      int i130;
      i130 = i128 + (5 * i36);
      #pragma unroll
      for(nvfuser_index_t i37 = 0; i37 < 5; ++i37) {
        T2[(i130 + i37)] = 0;
      }
    }
  }
  NVFUSER_UPDATE_MAGIC_ZERO
  #pragma unroll
  for(nvfuser_index_t i35 = 0; i35 < 2; ++i35) {
    int i151;
    i151 = 3 * i35;
    #pragma unroll
    for(nvfuser_index_t i36 = 0; i36 < 1; ++i36) {
      int i152;
      i152 = 5 * i36;
      int i153;
      i153 = i151 + i152;
      int i304;
      i304 = 3 - i152;
      #pragma unroll
      for(nvfuser_index_t i37 = 0; i37 < 5; ++i37) {
        int i154;
        i154 = i153 + i37;
        if (((i37 + nvfuser_zero) < i304)) {
          T2[i154]
            = T1[i154]
            + (float) 1.00000000000000000e+00;
        }
      }
    }
  }
  NVFUSER_UPDATE_MAGIC_ZERO
  #pragma unroll
  for(nvfuser_index_t i40 = 0; i40 < 1; ++i40) {
    int i192;
    i192 = 5 * i40;
    int i338;
    i338 = 3 - i192;
    // Alias Allocation - register
    auto& T3 = T2;
    #pragma unroll
    for(nvfuser_index_t i38 = 0; i38 < 5; ++i38) {
      int i193;
      i193 = i192 + i38;
      #pragma unroll
      for(nvfuser_index_t i39 = 0; i39 < 2; ++i39) {
        T3[(i38 + (5 * i39))]
          = T2[(i193 + (3 * i39))]
          + (float) 1.00000000000000000e+00;
      }
    }
    #pragma unroll
    for(nvfuser_index_t i41 = 0; i41 < 2; ++i41) {
      int i212;
      i212 = 5 * i41;
      int i237;
      i237 = i192 + (3 * i41);
      #pragma unroll
      for(nvfuser_index_t i42 = 0; i42 < 5; ++i42) {
        int i214;
        i214 = i42 + nvfuser_zero;
        if ((i214 < i338)) {
          T4[(i237 + i214)]
            = T3[(i212 + i42)]
            + (float) 1.00000000000000000e+00;
        }
      }
    }
  }
  NVFUSER_UPDATE_MAGIC_ZERO
}
```

Reply:
Does it produce wrong results?

Reply:
Yes, it produces wrong result. Let's move the discussion of this issue here: https://github.com/csarofeen/pytorch/pull/2490

Reply:
Ah, the traversal pattern is different between `T3` and `T2`, so they can't be reused.

Reply:
Yes, correct.